index.rst

```diff
@@ -50,7 +50,696 @@
+| *VisitPatchJoin*                                             |
++============+========+========================================+
+| visit_id   | uint64 | NOT NULL REFERENCES Visit (visit_id)   |
++------------+--------+----------------------------------------+
+| patch_id   | uint64 | NOT NULL REFERENCES Patch (patch_id)   |
++------------+--------+----------------------------------------+
+
++------------+--------+----------------------------------------+
+| *VisitTractJoin*                                             |
++============+========+========================================+
+| visit_id   | uint64 | NOT NULL REFERENCES Visit (visit_id)   |
++------------+--------+----------------------------------------+
+| tract_id   | uint64 | NOT NULL REFERENCES Tract (tract_id)   |
++------------+--------+----------------------------------------+
+
+
+.. _cs_datasets:
+
+Datasets
+--------
+
+Because the :ref:`DatasetTypes <DatasetType>` in present in a
+:ref:`DataRepository` may vary from repository to repository, the
+:ref:`Dataset` tables in the Common Schema are defined dynamically according to
+a set of rules:
+
+ - There is a table for each :ref:`DatasetType`, with entries corresponding to
+   :ref:`Datasets <Dataset>` that are present in the :ref:`DataRepository` (and
+   only these).
+
+ - The name of the table should be the name of the :ref:`DatasetType`.
+
+ - The table has a foreign key field relating to each :ref:`DataUnit` table that
+   is used to label the :ref:`DatasetType`.
+
+ - The table has at least the following additional fields:
+
++------------+--------+---------------------------------------------+
+| dataset_id | uint64 | PRIMARY KEY REFERENCES Dataset (dataset_id) |
++------------+--------+---------------------------------------------+
+| uri        | str    |                                             |
++------------+--------+---------------------------------------------+
+
+The ``dataset_id`` field is both a primary key that must be unique across
+elements in this table and a link to the more general Dataset table described in
+the :ref:`Provenance <cs_Provenance>` section; this means that it must be
+globally unique across *all* dataset tables, virtually guaranteeing that these
+per-:ref:`DatasetType` tables will be implemented as views into a larger table.
+
+The ``uri`` field contains a string that can be used to local the file or other
+entity that contains the stored :ref:`Dataset`.  While this may be generated
+differently according to different butler configurations when the file is first
+written, after it is written we do not expect the name to change and hence
+record it in the database; this reduces the need for butler implementations to
+be aware of past configurations in addition to their current confirguration. For
+multi-file composite datasets, this field should be NULL, and another table
+(TBD) can be used to associate the composite with its leaf-node :ref:`Datasets
```

Comment:
There are trade-offs here. Since the design with `NULL` necessitates multiple queries (in the case you don't already know from the `DatasetType` if it will be composite or not. But always decoupling with a secondary table might require a more expensive query (extra join) then would be needed in most cases.